sos/report/plugins/system.py

```diff
@@ -47,6 +48,42 @@ def setup(self):
                                 "environment_varnames",
                                 plug_dir=True)
 
+        sys_proxy_content = self._check_system_proxy_setting()
+
+        self.add_cmd_output(f'echo "{sys_proxy_content}"',
+                            suggest_filename="system_proxy_setting",
+                            root_symlink="system-proxy-setting",
+                            tags="sys_proxy")
+
+    def _check_system_proxy_setting(self):
+        sys_proxy_env_vars = [
+            'http_proxy',
+            'https_proxy',
+            'HTTP_PROXY',
+            'HTTPS_PROXY',
+        ]
+        sys_proxy_settings = {
+            var: os.environ.get(var)
+            for var in sys_proxy_env_vars
+            if os.environ.get(var)
+        }
+
+        def scrub_username_password(url):
+            """
+            Replace username:password@ with ********@ in proxy URL if present
+            """
+            return re.sub(r'//([^/@]+)@', '//********@', url)
+
+        if sys_proxy_settings:
+            sys_proxy_content = "\n".join(
+                f"{var}={scrub_username_password(value)}"
+                for var, value in sys_proxy_settings.items()
+            )
+        else:
+            sys_proxy_content = "No system proxy settings found."
```

Comment:
No strong preference here either. Isnt empty output better? It's meaning is almost the same like with this text, but it allows easier tooling which can rely on "every line has a proxy setting, no comments"?

But some people can see an empty file as confusing..?

Reply:
seems fine for empty and up to users. I see `/etc/environment` always empty.